src/Page.js

```diff
@@ -658,10 +657,8 @@ class Page {
     const footerContent = fs.readFileSync(footerPath, 'utf8');
     // Set footer file as an includedFile
     this.includedFiles.add(footerPath);
-    // Map variables
-    const parentSite = urlUtils.getParentSiteAbsolutePath(this.sourcePath, this.rootPath, this.baseUrlMap);
-    const userDefinedVariables = this.userDefinedVariablesMap[parentSite];
-    return `${pageData}\n${njUtil.renderRaw(footerContent, userDefinedVariables)}`;
+
+    return `${pageData}\n${this.variablePreprocessor.renderSiteVariables(this.sourcePath, footerContent)}`;
```

Comment:
example of site variable processing boilerplate reduction (the second arrow), similar ones follow throughout in `page.js`